glitter/publisher/admin.py

```diff
@@ -0,0 +1,48 @@
+from __future__ import unicode_literals
+
+from django import forms
+from django.contrib.contenttypes.admin import GenericStackedInline
+
+from .forms import object_version_choices
+from .models import PublishAction
+
+
+class ActionInline(GenericStackedInline):
+    model = PublishAction
+    fields = ('scheduled_time', 'publish_version')
+    extra = 0
+
+    def get_formset(self, request, obj=None, form=None, **kwargs):
+        class VersionForm(forms.ModelForm):
```

Comment:
I think we discuss this previously but I prefer going this way:

```        formset = super(ActionInline, self).get_formset(request, obj, **kwargs)
        formset.form.base_fields['publish_version'].widget = forms.widgets.Select(
            choices=object_version_choices(obj=obj)
        )
        formset.request = request
        
        return formset
```
It's just 4 lines also you need to set at the top `formset = ActionInlineFormset` and create form set with the `save_new` and `save_existing` methods hope this helps.